odoo/tools/translate.py

```diff
@@ -842,10 +842,20 @@ def _extract_translatable_qweb_terms(element, callback):
                 and "t-js" not in el.attrib
                 and not ("t-jquery" in el.attrib and "t-operation" not in el.attrib)
                 and el.get("t-translation", '').strip() != "off"):
+
             _push(callback, el.text, el.sourceline)
-            for att in ('title', 'alt', 'label', 'placeholder'):
-                if att in el.attrib:
-                    _push(callback, el.attrib[att], el.sourceline)
+            # Do not export terms contained on the Component directive of Owl
+            # attributes in this context are most of the time variables,
+            # not real HTML attributes.
+            # Node tags starting with a capital letter are considered owl Components
+            # and a widespread convention and good practice for DOM tags is to write
+            # them all lower case.
+            # https://www.w3schools.com/html/html5_syntax.asp
+            # https://github.com/odoo/owl/blob/master/doc/reference/component.md#composition
+            if not el.tag[0].isupper() and 't-component' not in el.attrib:
```

Comment:
https://docs.python.org/3/library/stdtypes.html#str.isupper

When I read the doc of `isupper` I don't understand how this works, it only returns true if all char in the string are upper? Don't we want also to catch if any letter is upper? Which actually does in the test, I'm confused.

Reply:
I just take the first letter by doing tag[0] just before

Reply:
I missed that part indeed, that makes sense now.